src/go/rpk/pkg/cli/cmd/topic/delete.go

```diff
@@ -64,17 +67,35 @@ For example,
 
 			resps, err := adm.DeleteTopics(context.Background(), topics...)
 			out.MaybeDie(err, "unable to issue delete topics request: %v", err)
-			tw := out.NewTable("topic", "status")
-			defer tw.Flush()
+
+			topicsCollection := topicsCollection{}
+			var deleteError bool
 			for _, t := range resps.Sorted() {
 				msg := "OK"
 				if t.Err != nil {
+					// If error msg is not okay, set bool so structured output doesn't require consumer to parse message to detect errors.
+					deleteError = true
 					msg = t.Err.Error()
 				}
-				tw.Print(t.Topic, msg)
+				topicsCollection.AddTopic(topic{
+					Name:    t.Topic,
+					Message: msg,
+					Error:   deleteError,
+				})
+			}
+
+			if format != "text" {
+				out.StructredPrint[any](topicsCollection, format)
+			} else {
+				tw := out.NewTable("topic", "status")
+				defer tw.Flush()
+				for _, t := range topicsCollection.Topics {
+					tw.Print(t.Name, t.Message)
+				}
 			}
 		},
 	}
+	cmd.Flags().StringVar(&format, "format", "text", "Output format (text, json, yaml). Default: text")
```

Comment:
```suggestion
	cmd.Flags().StringVar(&format, "format", "text", "Output format (text, json, yaml).")
```
No need to add the "Default: text" it gets automatically added by cobra.

Reply:
I must have missed cobra adding it, will remove it from all the spots I added it.